source/strings.tex

```diff
@@ -1478,32 +1443,19 @@
 \begin{itemdescr}
 \pnum
 \effects
-If
-\tcode{*this}
-and \tcode{str} are not the same object,
-modifies
-\tcode{*this}
-as shown in Table~\ref{tab:strings.op=}.
-
-\pnum
-If
-\tcode{*this}
-and \tcode{str} are the same object,
-the member has no effect.
+If \tcode{*this} and \tcode{str} are the same object, the member has no effect.
+Otherwise, modifies \tcode{*this} as described below.
 
 \pnum
 \returns
 \tcode{*this}.
 
-\begin{libefftabvalue}
-{\tcode{operator=(const basic_string\&)} effects}
-{tab:strings.op=}
-\tcode{data()}      &
-points at the first element of an allocated copy of the array whose first
-element is pointed at by \tcode{str.data()}                                             \\
-\tcode{size()}      &   \tcode{str.size()}                                              \\
-\tcode{capacity()}  &   a value at least as large as \tcode{size()}                     \\
-\end{libefftabvalue}
+\pnum
+\postconditions
+\tcode{data()} points at the first element of an allocated copy
+of the array whose first element is pointed at by \tcode{str.data()},
+\tcode{size()} is equal to \tcode{str.size()}, and
+\tcode{capacity()} is a value at least as large as \tcode{size()}.
```

Comment:
I think the "If `*this` and `str` are the same object" condition should be moved down here -- the postcondition should ideally be written such that it applies for all (valid) calls to the function.

Reply:
Moved or copied? Do you want to delete the Effects clause?